### ISSUE TITLE
new  fresh deployment 

### ISSUE THREAD

**Issue by belvarr** [nokia/danm#96]
feature, support and help 

i face difficulty in deployment the DANM cni i try once and i have failed so i re-deploy new cluster and i am now in  step 2 in deployment section especially for kubeconfig.yaml and RBAC file  so would you please provide some explanation for creating those files 

also can i have some example for creating pod with cluster wide ipvlan

**Comment by Levovar** [comment]
There are almost ready-made examples for kubeconfig and CNI RBAC in https://github.com/nokia/danm/tree/master/integration/cni_config

**Comment by belvarr** [comment]
@Levovar thank you for your reply i already read them but i wanna some clear example and steps for creating the kubeconfig and the rbac for danm user

**Comment by Levovar** [comment]
but what would you like to see in addition to a literally spelled out exact manifests you would need to use? :)
The kubeconfig is a standard kubeconfig. Nothing DANM specific here, at all.
The RBAC settings are also standard Kubernetes APIs. You need to provision them, exactly this manifest, and then the DANM CNI will have all the required privileges.

I suggest formulating some exact questions regarding which part of the process is still unclear to you

**Comment by belvarr** [comment]
sorry for confusing you i am just new for kubernetes and i found DANM is cluster wide which that i need
 so i need to create kubeconfig for danm user is it right ?

**Comment by Levovar** [comment]
that's right. DANM CNI needs to know which user to use, and how to authenticate towards the K8s API server. that information it gets from the standard kubeconfig file.
in my cluster we have client certificate based authentication method, so that's what I put into the example. you need to adapt it to your own environment

once DANM binary can authenticate itself, the K8s API server will authenticate tis requests against the RBAC profile. this you can use as-is, all the required privileges are present in the example

**Comment by belvarr** [comment]
would you please confirm below step 
i have locate my kubeconfig file for my cluster then i took the <certificate-authority-data>,<client-certificate-data> and <client-key-data> and put it in DANM example_kubeconfig.yaml then rename the file to danmc.yml 
i have move the file to /etc/kubernetes/kubeconfig/danmc.yml pointing the "kubeconfig" and "kubeconfig" in 00-danm.conf to it

$cat /etc/cni/net.d/00-danm.conf
{
  "name": "meta_cni",
  "name_comment": "Mandatory parameter, but can be anything",
  "type": "danm",
  "type_comment": "danm",
  "kubeconfig": "/etc/kubernetes/kubeconfig/danmc.yml",
  "kubeconfig_comment": "/etc/kubernetes/kubeconfig/danmc.yml",
  "cniDir": "/etc/cni/net.d",
  "cniDir_comment": "/etc/cni/net.d",
  "namingScheme": "awesome",
  "namingScheme_comment": ""
}

after that i apply 
kubectl create -f integration/cni_config/danm_rbac.yaml

follow the steps 3,4,5,6 but in step 7 the pods of netwatcher is pending another question here should i create role and serviceAccount  for netwatcher ? 

i am really appreciate your support

**Comment by Levovar** [comment]
> my cluster then i took the "certificate-authority-data","client-certificate-data" and "client-key-data"

if your API server authenticates users using client certs signed by your CA, then yes

> rename the file to danmc.yml
i have move the file to /etc/kubernetes/kubeconfig/danmc.yml pointing the "kubeconfig" and "kubeconfig" in 00-danm.conf to it

the name of the file can be anything, just configure the exact path into the  DANM CNI config file
BTW the "_comment" fields are really just that, comments to explain the role of the attributes. I only used this in the schema because there is no way to comment attributes in JSON, but you can safely delete these lines

> after that i apply
kubectl create -f integration/cni_config/danm_rbac.yaml

yep

> follow the steps 3,4,5,6 but in step 7 the pods of netwatcher is pending another question here should i create role and serviceAccount for netwatcher ?

yep. Netwatcher, and also svcwatcher requires their own RBAC setting and service accounts, you can find the references here:
https://github.com/nokia/danm/blob/master/integration/manifests/netwatcher/0netwatcher_rbac.yaml
https://github.com/nokia/danm/blob/master/integration/manifests/svcwatcher/0svcwatcher_rbac.yaml
The absence of the RBAC roles cannot block the provisioning of the Pods though, that is probably related to something else. Check the logs to figure out why it is pending

**Comment by Levovar** [comment]
one more note: I hope you are deploying the code from the 3.3 release, and not from master. master is currently in the state of transition

**Comment by belvarr** [comment]
@Levovar Thank you for your kind support it is work finally i will add new comment for all steps
